[TEST] Prep 3.0.0  using vSphere API version 7.0.3

alinabuzachis opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Prep 3.0.0  using vSphere API version 7.0.3
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->- New Module Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
several

alinabuzachis commented:
@mariolenz @jillr Integration tests seem passing with the new new API version.

mariolenz commented:
Cool!

There are a lot of linter errors, tough. I think we can fix those `Imports are incorrectly sorted and/or formatted` by changing the order here:

https://github.com/ansible-community/ansible.content_builder/blob/1a8071e914aa1f25a84670b320c2295736a00818/roles/module_openapi_cloud/templates/module_directory/vmware_rest/header.j2#L30-L41

I'll create a PR for that and then maybe we can try again.

mariolenz commented:
@alinabuzachis Did the three new mappings I've suggested fix generating the modules? If yes, I wouldd add them to my PR.

mariolenz commented:
@alinabuzachis @jillr ansible-community/ansible.content_builder#74

Once this one and ansible-community/ansible.content_builder#73 are merged, I think we should give it another try 😃

alinabuzachis commented:
> @alinabuzachis Did the three new mappings I've suggested fix generating the modules? If yes, I would add them to my PR.

Yes, they fixed.

alinabuzachis commented:
> [ansible-community/ansible.content_builder#74](https://github.com/ansible-community/ansible.content_builder/pull/74)

Ah, I see now https://github.com/ansible-community/ansible.content_builder/pull/74

mariolenz reviewed (commented):
@alinabuzachis There are still a lot of failing CI jobs, and I think we should fix them before merging. I'm sure ansible-community/ansible.content_builder#73 and ansible-community/ansible.content_builder#74 would, but we need someone to merge them. I've tested in #452 and this looked OK to me. Of course, you can patch my changes into your local content_builder but having them upstream would be better.

One other thing: I've removed all module documentation from the [community.vmware](https://github.com/ansible-collections/community.vmware) repository. It just felt like a waste of resources to generate them and drag them around. The stuff is available in the [Ansible community documentation](https://docs.ansible.com/ansible/latest/collections/community/vmware/index.html), anyway... and [your module documentation](https://docs.ansible.com/ansible/latest/collections/vmware/vmware_rest/index.html#plugins-in-vmware-vmware-rest) is, too. Maybe you should consider this also. Just an idea ;-)

Since this doesn't concern the published collection, I don't think this would need a new major release and you could do it anytime. But preparing one, anyway, seems like a good moment to do it. That is, if you want to do it... as I've said: Just an idea.

alinabuzachis commented:
> @alinabuzachis There are still a lot of failing CI jobs, and I think we should fix them before merging. I'm sure [ansible-community/ansible.content_builder#73](https://github.com/ansible-community/ansible.content_builder/pull/73) and [ansible-community/ansible.content_builder#74](https://github.com/ansible-community/ansible.content_builder/pull/74) would, but we need someone to merge them. I've tested in #452 and this looked OK to me. Of course, you can patch my changes into your local content_builder but having them upstream would be better.
> 
> One other thing: I've removed all module documentation from the [community.vmware](https://github.com/ansible-collections/community.vmware) repository. It just felt like a waste of resources to generate them and drag them around. The stuff is available in the [Ansible community documentation](https://docs.ansible.com/ansible/latest/collections/community/vmware/index.html), anyway... and [your module documentation](https://docs.ansible.com/ansible/latest/collections/vmware/vmware_rest/index.html#plugins-in-vmware-vmware-rest) is, too. Maybe you should consider this also. Just an idea ;-)
> 
> Since this doesn't concern the published collection, I don't think this would need a new major release and you could do it anytime. But preparing one, anyway, seems like a good moment to do it. That is, if you want to do it... as I've said: Just an idea.

Fixed linters and sanity errors! I tested too and those patches look good to me! Thanks!

We have the same way of handing docs in other collections too (see amazon.aws https://docs.ansible.com/ansible/latest/collections/amazon/aws/index.html). I'll try to add the Github workflow in another PR and eventually merged it after the 3.0.0 release. We can always do minor releases.

mariolenz reviewed (approved):
LGTM

content_builder has created new sanity ignores for 2.9 to 2.13. We don't need them anymore, but I don't think we have to remove them now. We can do this later, no problem.

Maybe someone else should also have a look at this. Nearly 300 changed files is quite a lot and it's easy to overlook something.

mariolenz commented:
:partying_face:
